packages/myst-cli/src/transforms/images.ts

```diff
@@ -147,6 +151,143 @@ export async function transformImages(
   );
 }
 
+type ConversionFn = (session: ISession, svg: string, writeFolder: string) => Promise<string | null>;
+
+export async function transformImageFormats(
+  session: ISession,
+  mdast: Root,
+  file: string,
+  writeFolder: string,
+  opts?: { altOutputFolder?: string; imageExtensions?: string[] },
+) {
+  const images = selectAll('image', mdast) as GenericNode[];
+  // Ignore all images with supported extension types
+  const validExts = opts?.imageExtensions ?? DEFAULT_IMAGE_EXTENSIONS;
+  const unsupportedImages = images.filter((image) => {
+    return !validExts.includes(path.extname(image.url));
+  });
+  if (unsupportedImages.length === 0) return;
+
+  // Gather unsupported SVG & GIF images that may be converted to supported types
```

Comment:
All this logic of deciding which unsupported image types may be converted to a different supported type (and whether to use imagemagick or inkscape) is already quite complex and bloated. Likely any additional format handling will require us to refactor this...